Automattic/wp-calypso
# Site Assembler - Refactoring to align the pattern data with PTK API response

Related to #74957

https://user-images.githubusercontent.com/1881481/229983770-d7a3741a-6717-421c-b883-0977b63b21b8.mov


## Proposed Changes

**These changes are not changing anything on the user's end.**

In preparation for adding [All Dotcom patterns](https://github.com/Automattic/wp-calypso/issues/74926): 

- Refactoring to align the pattern data with the PTK API response


## Testing Instructions



Everything should work as expected. Nothing is changing on the user's end.

* Use the `Calypso live` link on the first comment
* Create a new site and continue until the Design Picker
* Click `Start designing` from the bottom of the design picker 
* Choose a header, then `Save`
* Choose a footer, then `Save`
* Click `Sections` 
* Click `Add patterns` to explore the categories and add patternst
* Replace, order, and delete patterns
* Click `Continue` and verify the patterns are added to a Home template on your site

## Pre-merge Checklist
